sdk/azcore/internal/exported/exported.go

```diff
@@ -0,0 +1,143 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package exported
+
+import (
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared"
+)
+
+// TokenRequestOptions contain specific parameter that may be used by credentials types when attempting to get a token.
+// Exported as policy.TokenRequestOptions.
+type TokenRequestOptions struct {
+	// Scopes contains the list of permission scopes required for the token.
+	Scopes []string
+	// TenantID contains the tenant ID to use in a multi-tenant authentication scenario, if TenantID is set
+	// it will override the tenant ID that was added at credential creation time.
+	TenantID string
+}
+
+// TokenCredential represents a credential capable of providing an OAuth token.
+// Exported as azcore.TokenCredential.
+type TokenCredential interface {
```

Comment:
Should we just define this in azcore? That doesn't appear to create a cycle

Reply:
We could.  In fact, looks like there might be a few others too (I'll play around with it).  My only concern is if this might cause future hardships.  I've been trying to keep the dependencies flowing one way, i.e. parents can reach into children and cross-sibling (there are a few exceptions).  Maybe it will be ok.

Reply:
The other cases introduce cycles, so this looks to be the only case.

I'm tempted to leave this as-is to avoid introducing more cycles.

Reply:
I lean toward defining things where they're exported whenever possible, to get better documentation and code completion. Is moving a definition behind an alias breaking in some way?

Reply:
No, it would be compatible.  We'd just have to undo the change that directly exports it (touches 11 files).  Not terrible but might not be obvious at first glance.

Reply:
Fixed